src/nominatim_api/results.py

```diff
@@ -225,23 +220,44 @@ def lon(self) -> float:
         """
         return self.centroid[0]
 
+    @property
+    def display_name(self) -> Optional[str]:
+        """ Dynamically compute the display name for the result place
+            and, if available, its address information..
+        """
+        # No longer sets the locale name, want to do that explicitly in format.py
+        # using the formatter object
+        if self.address_rows:  # if this is true we need additional processing
+            label_parts: List[str] = []
+
+            for line in self.address_rows:  # assume locale_name is set by external formatter
+                line.local_name = line.local_name or line.display_name  # fallback to populate
```

Comment:
This is still not right. You must not set the `local_name` field as a side effect of this function. Use a local variable:

```
if line.address and line.names:
    address_name = line.display_name

    if address_name and (not label_parts or label_parts[-1] != address_name):
        label_parts.append(address_name)
```